apps/omg/lib/omg/state.ex

```diff
@@ -222,4 +224,17 @@ defmodule OMG.State do
     :ok = OMG.Bus.broadcast("events", {:preprocess_emit_events, event_triggers})
     :ok = OMG.Bus.direct_local_broadcast("blocks", {:enqueue_block, block})
   end
+
+  defp load_necessary_utxos_from_db(state, tx) do
+    tx
+    |> Transaction.get_inputs()
```

Comment:
I think everything except `Enum.map(&utxo_from_db/1)` should go to `core.ex` and have tests